build_tools/circle/build_doc.sh

```diff
@@ -159,21 +159,20 @@ if [[ `type -t deactivate` ]]; then
   deactivate
 fi
 
-MAMBAFORGE_PATH=$HOME/mambaforge
-# Install dependencies with mamba
-wget -q https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-x86_64.sh \
-    -O mambaforge.sh
-chmod +x mambaforge.sh && ./mambaforge.sh -b -p $MAMBAFORGE_PATH
-export PATH="/usr/lib/ccache:$MAMBAFORGE_PATH/bin:$PATH"
-
+# Install Miniforge
+MINIFORGE_URL="https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh"
+curl -L --retry 10 $MINIFORGE_URL -o miniconda.sh
+MINIFORGE_PATH=$HOME/miniforge3
+bash ./miniconda.sh -b -p $MINIFORGE_PATH
+source $MINIFORGE_PATH/etc/profile.d/conda.sh
+conda activate
```

Comment:
I seem to remember that you had issues with updating the lock-files though with micromamba, right, and that the fix for you was `micromamba install mamba` right? My wild guess is that there were issues with virtual packages that were fixed in conda (I did one of the PR by the way). Because micromamba reimplements these parts, they were issues, there is a small chance that someone reported it and that it was fixed.

I am not convinced the "lot smaller" this matters that much in practice for the, CI maybe you save a few seconds for the downloads and 2-3 lines for the installation code.

All in all, I would be in favour of being conservative here.

Reply:
The solution for me was to install both `mamba` and `conda` to be able to update lock files.

The smaller thing is significant since `micromamba` doesn't come with a base environment. So there's one whole environment saved by not using `conda` here.

Reply:
I think this PR is already a net improvement. Can we discuss `micromamba` in a follow up PR?